php-frontend/src/main/java/org/sonar/php/tree/impl/lexical/InternalSyntaxToken.java

```diff
@@ -80,8 +84,23 @@ public boolean is(TokenType type) {
   }
 
   @Override
-  public boolean is(Kind... kind) {
-    throw new UnsupportedOperationException("Not implemented yet");
+  public Kind getKind() {
+    return Kind.TOKEN;
+  }
+
+  @Override
+  public Iterator<Tree> childrenIterator() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public boolean isLeaf() {
+    return true;
+  }
+
+  @Override
+  public void accept(TreeVisitor visitor) {
+    // FIXME do nothing at the moment
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)